packages/react-native-worklets/src/shareables.ts

```diff
@@ -253,6 +253,36 @@ function cloneArray<T extends unknown[]>(
   shouldPersistRemote: boolean,
   depth: number
 ): ShareableRef<T> {
+  if (value.length > 20) {
+    const isArrayOfStrings = value.every(
+      (element) => typeof element === 'string'
+    );
+    const isArrayOfNumbers = value.every(
+      (element) => typeof element === 'number'
+    );
+    if (isArrayOfStrings) {
+      const clone = WorkletsModule.makeShareableArrayOfStrings(
+        value as string[]
+      );
+      shareableMappingCache.set(value, clone);
+      shareableMappingCache.set(clone);
+
+      freezeObjectInDev(value);
+      return clone as ShareableRef<T>;
```

Comment:
Do we still need `as ShareableRef<T>` here?

Reply:
yes